lib/charms/layer/apache_kafka.py

```diff
@@ -14,14 +16,29 @@ def __init__(self, dist_config=None):
         self.resources = {
             'kafka': 'kafka-%s' % utils.cpu_arch(),
         }
-        self.verify_resources = utils.verify_resources(*self.resources.values())
+
+    def verify_resources(self):
+        try:
+            return hookenv.resource_get('kafka')
+        except NotImplementedError:
+            return utils.verify_resources(*self.resources.values())()
 
     def install(self):
         self.dist_config.add_users()
         self.dist_config.add_dirs()
-        jujuresources.install(self.resources['kafka'],
-                              destination=self.dist_config.path('kafka'),
-                              skip_top_level=True)
+        try:
+            filename = hookenv.resource_get('kafka')
+            extracted = fetch.install_remote('file://' + filename)
+            # get the nested dir
+            extracted = os.path.join(extracted, os.listdir(extracted)[0])
+            destination = self.dist_config.path('kafka')
+            if os.path.exists(destination):
+                shutil.rmtree(destination)
+            shutil.copytree(extracted, destination)
```

Comment:
this feels uber clumsy..  are we really calling fetch.install_remote to extract in one dir only to copy it to another?  why not just extract to the destination in the first place?